src/main/java/com/artipie/auth/GithubAuth.java

```diff
@@ -64,11 +67,23 @@ public GithubAuth() {
         Optional<Authentication.User> result = Optional.empty();
         final Matcher matcher = GithubAuth.PTN_NAME.matcher(username);
         if (matcher.matches()) {
-            final String login = this.github.apply(password).toLowerCase(Locale.US);
-            if (
-                Objects.equals(login, matcher.group(1).toLowerCase(Locale.US))
-            ) {
-                result = Optional.of(new Authentication.User(login));
+            try {
+                final String login = this.github.apply(password).toLowerCase(Locale.US);
+                if (
+                    Objects.equals(login, matcher.group(1).toLowerCase(Locale.US))
+                ) {
+                    result = Optional.of(new Authentication.User(login));
+                }
+            } catch (final AssertionError error) {
+                if (
+                    new Unchecked<>(
+                        new Not(
+                            new Contains(error.getMessage(), "401 Unauthorized")
+                        )
+                    ).value()
+                ) {
+                    throw error;
```

Comment:
I think throwing `AssertionError` is not really a good place for production code with disabled assertions (`java -ea` flag is not enabled). Maybe we can re-throw it with some different type?

Reply:
Ok